Source/ACE/Entity/Player.cs

```diff
@@ -1633,20 +1692,157 @@ public void HandleActionPutItemInContainer(ObjectGuid itemGuid, ObjectGuid conta
+
+                if (PhysicsData.CSetup != null && item.ClothingBaseEffects.ContainsKey((uint)PhysicsData.CSetup))
+                // Check if the player model has data. Gear Knights, this is usually you.
+                {
+                    // Add the model and texture(s)
+                    ClothingBaseEffect clothingBaseEffec = item.ClothingBaseEffects[(uint)PhysicsData.CSetup];
+                    foreach (CloObjectEffect t in clothingBaseEffec.CloObjectEffects)
+                    {
+                        byte partNum = (byte)t.Index;
+                        ModelData.AddModel((byte)t.Index, (ushort)t.ModelId);
+                        coverage.Add(partNum);
+                        foreach (CloTextureEffect t1 in t.CloTextureEffects)
+                            ModelData.AddTexture((byte)t.Index, (ushort)t1.OldTexture, (ushort)t1.NewTexture);
+                    }
+
+                    PhysicsData.ItemsEquipedCount += 1;
+                    foreach (ModelPalette p in wo.ModelData.GetPalettes)
+                        ModelData.AddPalette(p.PaletteId, p.Offset, p.Length);
+                }
+            }
+            // Add the "naked" body parts. These are the ones not already covered.
+            if (PhysicsData.CSetup != null)
+            {
+                SetupModel baseSetup = SetupModel.ReadFromDat((uint)PhysicsData.CSetup);
+                for (byte i = 0; i < baseSetup.SubObjectIds.Count; i++)
+                {
+                    if (!coverage.Contains(i) && i != 0x10) // Don't add body parts for those that are already covered. Also don't add the head, that was already covered by AddCharacterBaseModelData()
+                        ModelData.AddModel(i, baseSetup.SubObjectIds[i]);
+                }
+            }
+        }
+
+        public void HandleActionWieldItem(Container container, uint itemId, uint location)
+        {
+            ObjectGuid itemGuid = new ObjectGuid(itemId);
+            if (GetInventoryItem(itemGuid) == null)
+            {
+                HandlePickup(container, itemGuid, location, PropertyInstanceId.Wielder);
+                return;
+            }
+            // Ok do you have this - if not bail
+            if (GetInventoryItem(itemGuid) == null)
+                return;
+            UpdateAppearance(container, itemId);
+            Session.Network.EnqueueSend(new GameMessageSound(Guid, Sound.WieldObject, (float)1.0),
+                                        new GameMessageObjDescEvent(this),
+                                        new GameMessageUpdateInstanceId(container.Guid, itemGuid, PropertyInstanceId.Wielder),
+                                        new GameEventWieldItem(Session, itemGuid.Full, location));
+            CurrentLandblock.EnqueueBroadcast(Location, Landblock.MaxObjectRange, new GameMessageObjDescEvent(this));
+        }
+
+        public void HandleActionPutItemInContainer(ObjectGuid itemGuid, ObjectGuid containerGuid, uint location = 0)
```

Comment:
Same as above -- move the work into an action, or this will race and crash the server once we start getting a larger workload.